models/org_team.go

```diff
@@ -34,6 +35,64 @@ type Team struct {
 	Units       []*TeamUnit `xorm:"-"`
 }
 
+// SearchTeamOptions holds the search options
+type SearchTeamOptions struct {
+	UserID      int64
+	UserIsAdmin bool
+	Keyword     string
+	OrgID       int64
+	IncludeDesc bool
+	Limit       int
+}
+
+// SearchTeam search for teams. Caller is responsible to check permissions.
+func SearchTeam(opts *SearchTeamOptions) ([]*Team, int64, error) {
+	if opts.Limit <= 0 {
+		opts.Limit = 10
+	}
+	var cond = builder.NewCond()
+
+	if len(opts.Keyword) > 0 {
+		lowerKeyword := strings.ToLower(opts.Keyword)
+		var keywordCond = builder.NewCond()
+		if opts.IncludeDesc {
+			keywordCond = builder.Or(
+				builder.Like{"lower_name", lowerKeyword},
+				builder.Like{"LOWER(description)", lowerKeyword},
+			)
+		} else {
+			keywordCond = builder.Or(
+				builder.Like{"lower_name", lowerKeyword},
+			)
+		}
+		cond = cond.And(keywordCond)
+	}
+
+	cond = cond.And(builder.Eq{"org_id": opts.OrgID})
+
+	sess := x.NewSession()
+	defer sess.Close()
+
+	count, err := sess.
+		Where(cond).
+		Count(new(Team))
+
+	if err != nil {
+		return nil, 0, fmt.Errorf("Count: %v", err)
```

Comment:
Just return original error:
```suggestion
		return nil, 0, err
```

Reply:
Fixed